### ISSUE TITLE
"Error: SSL/TLS required on the control channel"

### ISSUE THREAD

**Issue by Jany-M** [Natizyskunk/vscode-sftp#73]
I get this same issue with liximomo's version, but he's not maintaining his extension anymore, so I'm asking you.

The error I get is **Error: SSL/TLS required on the control channel**, on the profile I'm using it's set to ftp, port 21 (which is correct, sftp is not required by server for this user and connection).

```
[10-18 13:53:14] [error] Error: SSL/TLS required on the control channel
	at D (c:\Users\janym\.vscode\extensions\natizyskunk.sftp-1.15.4\dist\extension.js:61:104522)
	at p.<anonymous> (c:\Users\janym\.vscode\extensions\natizyskunk.sftp-1.15.4\dist\extension.js:61:105246)
	at p.emit (events.js:315:20)
	at p._write (c:\Users\janym\.vscode\extensions\natizyskunk.sftp-1.15.4\dist\extension.js:61:118167)
	at writeOrBuffer (internal/streams/writable.js:358:12)
	at p.Writable.write (internal/streams/writable.js:303:10)
	at Socket.d (c:\Users\janym\.vscode\extensions\natizyskunk.sftp-1.15.4\dist\extension.js:61:107618)
	at Socket.emit (events.js:315:20)
	at addChunk (internal/streams/readable.js:309:12)
	at readableAddChunk (internal/streams/readable.js:284:9)
	at Socket.Readable.push (internal/streams/readable.js:223:10)
	at TCP.onStreamRead (internal/stream_base_commons.js:188:23) upload j:\some_file.php
```

**Comment by Natizyskunk** [comment]
I'll try to take a look at it.

**Comment by Rasalas** [comment]
I had the same issue. (~same connection type, same error message).
I'm adding info here, if someone else has that problem.

## Error
While connecting I got the following error message
![image](https://github.com/Natizyskunk/vscode-sftp/assets/7483565/f705a8cb-8d97-4c96-8473-feacceff69c8)

with the following terminal output
```
[01-21 23:27:15] [error] Error: SSL/TLS required on the control channel
    at g (c:\Users\tbuck\.vscode\extensions\natizyskunk.sftp-1.16.3\dist\extension.js:2:28508)
    at p.<anonymous> (c:\Users\tbuck\.vscode\extensions\natizyskunk.sftp-1.16.3\dist\extension.js:2:29232)
    at p.emit (node:events:513:28)
    at p.emit (node:domain:489:12)
    at p._write (c:\Users\tbuck\.vscode\extensions\natizyskunk.sftp-1.16.3\dist\extension.js:2:42069)
    at writeOrBuffer (node:internal/streams/writable:392:12)
    at _write (node:internal/streams/writable:333:10)
    at p.Writable.write (node:internal/streams/writable:337:10)
    at Socket.d (c:\Users\tbuck\.vscode\extensions\natizyskunk.sftp-1.16.3\dist\extension.js:2:31604)
    at Socket.emit (node:events:513:28)
    at Socket.emit (node:domain:489:12)
    at addChunk (node:internal/streams/readable:324:12)
    at readableAddChunk (node:internal/streams/readable:297:9)
    at Socket.Readable.push (node:internal/streams/readable:234:10)
    at TCP.onStreamRead (node:internal/stream_base_commons:190:23) 
```
## Solution

This problem probably happened to me, because my host doesn't allow raw FTP connections anymore.

I tried a full on FTP connection (without SSL/TLS) with FileZilla and got the same error response:
```
Response: 550 SSL/TLS required on the control channel
```
[In this issue of liximomo's version](https://github.com/liximomo/vscode-sftp/issues/872) I read that I need to add `"secure": true` to my sftp.json.


I then got the following error:
```
Hostname/IP does not match certificate's altnames: Host: example.com. is not in the cert's altnames:
DNS:*ftp.example.com, DNS:ftp.example.com, 
...
```
Others had problems with self signed certificates.

Adding `"secureOptions": {"rejectUnauthorized": false}` to the sftp.json solves both those problems (be sure you know what that means).

## TL;DR
Try to add these lines to your sftp.json
```json
{
    //...
    "secure": true,
    "secureOptions": {
        "rejectUnauthorized": false
    },
    //...
}
```

(I don't know if this was OP's Problem [I can only assume], but this can probably be closed if you don't plan on writing that into the documentation/FAQs or to add a more specific error message for this type of error [The FTP connection answered with 550 SSL/TLS required on the control channel]).

**Comment by Jany-M** [comment]
@Rasalas that actually worked!!! thank you
